apps/web/src/components/forms/status-page-form.tsx

```diff
@@ -92,12 +93,11 @@ export function StatusPageForm({
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [checkUniqueSlug]);
 
-  // FIXME: 🐛 This is causing a bug
-  // useEffect(() => {
-  //   if (!watchSlug) {
-  //     form.setValue("slug", slugify(watchTitle));
-  //   }
-  // }, [watchTitle, form, watchSlug]);
+  useEffect(() => {
+    if (!watchSlug) {
+      form.setValue("slug", slugify(watchTitle));
+    }
+  }, [watchTitle, form, watchSlug]);
```

Comment:
Thanks for tackling that issue! 🚀

This will trigger the `.setValue` function only if the `slug` is undefined or an empty string in our case. If the user deletes the entire slugified content, the function will get fired and automatically resets the `slug` from `undefined` to the slugified content.

I think we should track if the `defaultValue.slug` exists. Only if not, we watch the changes from the `title` input to update the `slug` input. Changing the `slug` afterwards is possible as the `useEffect` will not get fired. Only if, on creation (as `defaultValues` is undefined)m the `title` is changed again after having updated the auto-generated `slug`, it will re-change the already updated `slug`. (Getting inception vibes here.)

Let me know if that works!

```suggestion
  useEffect(() => {
    if (!defaultValues?.title) {
      form.setValue("slug", slugify(watchTitle));
    }
  }, [watchTitle, form, defaultValues]);
```